components/net/blob_loader.rs

```diff
@@ -0,0 +1,91 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use hyper::header::{DispositionType, ContentDisposition, DispositionParam};
+use hyper::header::{Headers, ContentType, ContentLength, Charset};
+use hyper::http::RawStatus;
+use ipc_channel::ipc::{self, IpcSender};
+use mime::{Mime, Attr};
+use mime_classifier::MIMEClassifier;
+use net_traits::ProgressMsg::Done;
+use net_traits::blob_url_store::{parse_blob_url, BlobURLStoreEntry, BlobURLStoreError, BlobURLStoreMsg};
+use net_traits::response::HttpsState;
+use net_traits::{LoadConsumer, LoadData, Metadata, NetworkError};
+use resource_thread::{CancellationListener, send_error, start_sending_sniffed_opt};
+use std::str;
+use std::sync::Arc;
+
+
+// TODO: Check on GET
+// https://w3c.github.io/FileAPI/#requestResponseModel
+
+pub fn factory(load_data: LoadData,
+               start_chan: LoadConsumer,
+               classifier: Arc<MIMEClassifier>,
+               _cancel_listener: CancellationListener,
+               blob_url_store: IpcSender<BlobURLStoreMsg>) {
+    match parse_blob_url(&load_data.url) {
+        None => {
+            let format_err = NetworkError::Internal(format!("Invalid blob URL format {:?}", load_data.url));
+            send_error(load_data.url.clone(), format_err, start_chan);
+        }
+        Some((uuid, _fragment)) => {
+            let (tx, rx) = ipc::channel().unwrap();
+            let _ = blob_url_store.send(BlobURLStoreMsg::Request(uuid, tx)).unwrap();
```

Comment:
As I observe, it is more natural to let threads in net task to handle the requests (instead of directly consume it inside script), so I need to somehow message script thread to fetch the content of blob. Not sure if it makes a lot of sense